### PR TITLE
Add option to skip archive.

### PR DESCRIPTION
**Pull request by jorinvo**
See #243.
Upload binaries directly instead of creating archive.
This option, however currently doesn't work together with brew and fpm
since they both require archived artifacts.

I'm not that happy that my current solution doesn't work together with brew and fpm, 
@caarlos0 do you have an idea how to generate archived + naked versions side by side so we can use the archived ones for brew/fpm and the other ones as artifacts?

### CODE REVIEW

File: pipeline/archive/archive.go [hunk 1]
```diff
@@ -73,6 +76,13 @@ func create(ctx *context.Context, platform, name string) error {
 	return nil
 }
 
+func skip(ctx *context.Context, platform, name string) error {
+	log.Println("Skip archiving")
```
Comment (caarlos0) [inline pipeline/archive/archive.go]: maybe add the platform and/or name to the log as well?

**Comment by caarlos0** [comment]
I don't see why codecov is complaining about coverage... seems to me that all new code is covered...

**Comment by jorinvo** [comment]
I also don't get it.

**Comment by caarlos0** [comment]
ok, let's just wait for @client9, so we can have a PR prepared on godownloader as well :)

**Comment by jorinvo** [comment]
@caarlos0 any thoughts regarding brew/fpm support?

**Comment by caarlos0** [comment]
@jorinvo I think fpm may be kind of easy, no idea about how to do it with homebrew though...

**Comment by caarlos0** [comment]
@jorinvo most formulae in the brew repos build/compile the software they install, e.g.: https://github.com/Homebrew/homebrew-core/blob/master/Formula/shfmt.rb

**Comment by jorinvo** [comment]
@caarlos0 I like how our brew formulas don't depend on the `go` tool though..

Currently brew also only works if a `darwinamd64` build is included right?
We could also remove that dependency and always build one version specifically for brew.
Then there would be no conflict with the `skip` archive option either.

**Comment by caarlos0** [comment]
@jorinvo I thinks it's strange to build for darwin if the user didn't added it to the list of targets.

as far as I remember there is a warning or something when brew is enabled and there is no darwin builds, we may make it a failure while validating the config instead so it would be clearer...

**Comment by jorinvo** [comment]
@caarlos0 yes, you're right. 
Maybe we also don't need to support brew/fpm together with `archive: skip: true` for now until someone actually uses it.

**Comment by caarlos0** [comment]
@jorinvo  Agreed!

**Comment by client9** [comment]
just saw this... will look tomorrow am.

**Comment by client9** [comment]
Ah added to ticket, not PR.

Don't know much about brew.

I think `skip: true` is confusing should be `format: binary` if I understand the issue correct.

**Comment by jorinvo** [comment]
@client9 good idea using the existing `format` option. I will look into it tomorrow.

**Comment by jorinvo** [comment]
@client9 I updated it to use `format` instead of `skip`.

**Comment by caarlos0** [comment]
LGTM, @client9 what do you think?

**Comment by caarlos0** [comment]
@jorinvo I changed the base branch to `multiple-builds` and I'm doing the changes so both features can work together :)

**Comment by caarlos0** [comment]
For the future:

- create packages with fpm on naked releases
- create brew formulas for naked releases

**Pull request merged**